test/diric.jl

```diff
@@ -0,0 +1,19 @@
+# diric.jl Dirichlet kernel tests
+
+@testset "diric" begin
+    @test_throws ArgumentError diric(0, -2)
+    @test @inferred diric(0, 4) ≈ 1
+    @test @inferred diric(0 // 1, 5) == 1
+    @test @inferred diric(4π, 4) ≈ 1
+    @test @inferred diric(2π, 4) ≈ -1
+    @test @inferred diric(0, 5) ≈ 1
+    @test @inferred diric(2π, 5) ≈ 1
+    @test @inferred diric(π, 5) == 1 // 5
```

Comment:
```suggestion
    @test @inferred(diric(0, 4)) ≈ 1
    @test @inferred(diric(0 // 1, 5)) == 1
    @test @inferred(diric(4π, 4)) ≈ 1
    @test @inferred(diric(2π, 4)) ≈ -1
    @test @inferred(diric(0, 5)) ≈ 1
    @test @inferred(diric(2π, 5)) ≈ 1
    @test @inferred(diric(π, 5)) == 1 // 5
```
Otherwise, this verifies that the comparison can be inferred.

Reply:
Also, can we test a few arbitrary values which are not multiples of 2π?